src/BreakpointViewer.cpp

```diff
@@ -793,25 +789,19 @@ void BreakpointViewer::fillTableRow(int index, BreakpointRef::Type type, int row
 	}
 
 	// location
-	auto* item2 = table->item(row, LOCATION);
-	QString location;
+	if (type == BreakpointRef::BREAKPOINT) {
+		Symbol* s = debugSession.symbolTable().getAddressSymbol(bp.range->start);
 
-	if (bp.type == Breakpoint::BREAKPOINT) {
-		if (Symbol* s = debugSession.symbolTable().getAddressSymbol(item2->text())) {
-			if (s->value() == bp.range->start) {
-				location = s->text();
-			}
-		}
-	}
-	if (location.isEmpty()) {
-		int locLen = (bp.type == Breakpoint::WATCHPOINT_IOREAD
-		           || bp.type == Breakpoint::WATCHPOINT_IOWRITE) ? 2 : 4;
-		location = QString("%1%2%3").arg(hexValue(bp.range->start, locLen))
-		               .arg(bp.range->end ? ":" : "")
-		               .arg(bp.range->end ? hexValue(*bp.range->end, locLen) : "");
+		setTextField(type, row, LOCATION,
+		             s ? s->text() : locationString(bp.range->start, {}, 4),
+		             s ? locationString(bp.range->start, {}, 4) : "");
+	} else {
+		int adrLen = (bp.type == Breakpoint::WATCHPOINT_IOREAD || bp.type == Breakpoint::WATCHPOINT_IOWRITE)
+		           ? 2 : 4;
+		setTextField(type, row, LOCATION, locationString(bp.range->start, bp.range->end, adrLen));
```

Comment:
2 comments on this snippet (I'm not sure both comments can be addressed simultaneously).

* These lines 792-802 are fairly similar to the lines 405-413, can this logic be extracted into a helper function? Though maybe they are only similar *before* the review comments on line 407 is fixed (user-entered label name might change).
* I'm not sure which I like best: `locationString(bp.range->start, {}, 4)` or `hexValue(bp.range->start, 4)`. I see the appeal of using `locationString()` everywhere (especially if that would allow to factor out code). But it also seems a bit weird to make this detour in a context where we know we're printing a breakpoint (no end-address, always 4 digits).